admin/app/components/badges/badge.js

```diff
@@ -43,4 +52,48 @@ export default class Badge extends Component {
         return 'l‘ensemble des acquis du target profile.';
     }
   }
+
+  @action
+  async updateBadge(event) {
+    event.preventDefault();
+
+    try {
+      const badge = await this.store.findRecord('badge', this.args.badge.id);
+      const updatedBadge = this._updateBadgeFields(badge);
+      updatedBadge.save();
+      this.notifications.success('Le résultat thématique a été mis à jour.');
+      this.editMode = false;
+    } catch (error) {
+      console.error(error);
+      this.notifications.error('Erreur lors de la mise à jour du résultat thématique.');
+    }
+  }
+
+  _updateBadgeFields(badge) {
+    badge.title = this.form.title;
+    badge.key = this.form.key;
+    badge.message = this.form.message;
+    badge.altMessage = this.form.altMessage;
+    return badge;
+  }
+
+  @action
+  cancel() {
+    this.toggleEditMode();
+  }
+
+  @action
+  toggleEditMode() {
+    this.editMode = !this.editMode;
+    if (this.editMode) {
+      this._initForm();
+    }
+  }
+
+  _initForm() {
+    this.form.title = this.args.badge.title;
+    this.form.key = this.args.badge.key;
+    this.form.message = this.args.badge.message;
+    this.form.altMessage = this.args.badge.altMessage;
```

Comment:
Est-ce qu'il ne faudrait pas récupérer les données fraiches avec `await this.store.findRecord('badge', this.args.badge.id);` si jamais on accède à cette page directement par URL ?

Reply:
Conclusion: On ne gère pas ce cas, non réalisé ailleurs et pas forcément très utile dans le cadre d'une application utilisée en interne.